server/house/Room.cc

```diff
@@ -1,49 +1,88 @@
-#include <iostream>
-#include "Room.h"
-
-namespace smart_home {
-    Room::Room() : type(LIVING_ROOM) {}
-
-    Room::Room(RoomType roomType) : type(roomType) {}
-
-    RoomType Room::getType() const {
-        return type;
-    }
-
-    void Room::addTemperatureSensor(const std::string& name) {
-        try {
-            sensors.emplace_back(std::make_shared<TemperatureSensor>(name));
-         } catch (const std::bad_alloc& e) {
-             std::cerr << "Failed to add temperature sensor due to memory allocation error: " << e.what() << std::endl;
-             exit(1);
-         }
-    }
-
-    void Room::addHumiditySensor(const std::string& name) {
-        try {
-            sensors.emplace_back(std::make_shared<HumiditySensor>(name));
-        } catch (const std::bad_alloc& e) {
-            std::cerr << "Failed to add humidity sensor due to memory allocation error: " << e.what() << std::endl;
-            exit(1);
-        }
-    }
-
-    void Room::addMonoxideSensor(const std::string& name) {
-        try {
-            sensors.emplace_back(std::make_shared<MonoxideSensor>(name));
-        } catch (const std::bad_alloc& e) {
-            std::cerr << "Failed to add monoxide sensor due to memory allocation error: " << e.what() << std::endl;
-            exit(1);
-        }
-    }
-
-    void Room::generateRandomSensorValues() const {
-        for (auto& sensor : sensors) {
-            sensor->generateRandomValue();
-        }
-    }
-
-    const std::vector<std::shared_ptr<Sensor>>& Room::getSensors() const {
-        return sensors;
-    }
-}
+#include <iostream>
+#include "Room.h"
+
+namespace smart_home {
+    Room::Room() : type(LIVING_ROOM) {}
+
+    Room::Room(RoomType roomType) : type(roomType) {}
+
+    RoomType Room::getType() const {
+        return type;
+    }
+
+    void Room::addTemperatureSensor(const std::string& name) {
```

Comment:
Again with the SOLID stuff :). Ideally you'd have only "addSensor" and "removeSensor" in your class, that takes a base class sensor pointer as parameter and adds/remove it from the vector. This way it would be compliant with Liskov and with Dependency inversion principles.
What happens if you need to add 5 more types of sensors? you have to change your Room class to add another 10 function.